File: pilhas-filas/python/is_balanced_stack1.py
from stack1 import Stack1


def is_balanced(text):
    stack = Stack1()
    for char in text:
        if char == '(':
            stack.push(char)
        elif char == ')':
            if stack.is_empty():
                return False
            stack.pop()
    return stack.is_empty()


print(is_balanced("(())()"))  # true
print(is_balanced("())("))  # false

File: pilhas-filas/javascript/is-balanced-stack1.js
import Stack1 from "./stack1.js";

function isBalanced(text) {
  let stack = new Stack1();
  for (let char of text) {
    if (char === "(") {
      stack.push(char);
    } else if (char === ")") {
      if (stack.isEmpty()) {
        return false;
      }
      stack.pop();
    }
  }
  return stack.isEmpty();
}

console.log(isBalanced("(())()")); // true
console.log(isBalanced("())(")); // false
